packages/firebase_messaging/lib/firebase_messaging.dart

```diff
@@ -3,13 +3,41 @@
 // found in the LICENSE file.
 
 import 'dart:async';
+import 'dart:io';
+import 'dart:ui';
 
 import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
 import 'package:meta/meta.dart';
 import 'package:platform/platform.dart';
 
 typedef Future<dynamic> MessageHandler(Map<String, dynamic> message);
 
+const String _backgroundName = 'plugins.flutter.io/android_fcm_background';
+
+void _fcmSetupCallback() {
+  const MethodChannel _channel =
+      MethodChannel(_backgroundName, JSONMethodCodec());
+
+  // Setup Flutter state needed for MethodChannels.
+  WidgetsFlutterBinding.ensureInitialized();
+
+  // This is where the magic happens and we handle background events from the
+  // native portion of the plugin.
+  _channel.setMethodCallHandler((MethodCall call) async {
```

Comment:
It seems like it would be more readable/consistent with other plugins to read a method off the `MethodCall` here.

Reply:
Done